OData/src/System.Web.OData/OData/Builder/StructuralTypeConfiguration.cs

```diff
@@ -268,21 +268,13 @@ public virtual PrimitivePropertyConfiguration AddProperty(PropertyInfo propertyI
             ValidatePropertyNotAlreadyDefinedInDerivedTypes(propertyInfo);
 
             // Remove from the ignored properties
-            if (RemovedProperties.Contains(propertyInfo))
+            if (RemovedProperties.Any(prop => prop.Name.Equals(propertyInfo.Name)))
```

Comment:
Does this work, if with property name aliasing, such as `[DataMember(Name="abc")]`?